openedx/core/djangoapps/discussions/views.py

```diff
@@ -0,0 +1,120 @@
+"""
+Handle view-logic for the djangoapp
+"""
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys import InvalidKeyError
+from rest_framework import serializers
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from openedx.core.lib.api.permissions import IsStaff
+
+from .models import DiscussionsConfiguration
+
+
+PROVIDER_FEATURE_MAP = {
+    'cs_comments_service': [
+        'discussion-page',
+        'embedded-course-sections',
+        'lti',
+        'wcag-2.1',
+    ],
+    'piazza': [
+        'discussion-page',
+        'lti',
+    ],
+}
+
+
+class DiscussionsConfigurationView(APIView):
+    """
+    Handle configuration-related view-logic
+
+    blah blah blah,
+    this all goes in the DRF summary
+    """
+    authentication_classes = (JwtAuthentication, SessionAuthenticationAllowInactiveUser)
+    permission_classes = (IsStaff,)
+
+    class Serializer(serializers.BaseSerializer):
+        """
+        Serialize configuration responses
+        """
+
+        def create(self, validated_data):
+            """
+            Create and save a new instance
+            """
+            raise NotImplementedError
+
+        def to_internal_data(self, data):
+            """
+            Transform the *incoming* primitive data into a native value.
+            """
+            raise NotImplementedError
+
+        def to_representation(self, instance) -> dict:
+            """
+            Serialize data into a dictionary, to be used as a response
+            """
+            payload = {
+                'context_key': str(instance.context_key),
+                'enabled': instance.enabled,
+                'features': {
+                    'discussion-page',
+                    'embedded-course-sections',
+                    'lti',
+                    'wcag-2.1',
+                },
+                'plugin_configuration': instance.plugin_configuration,
+                'providers': {
+                    'active': instance.provider_type or '',
+                    'available': {
+                        provider: {
+                            'features': PROVIDER_FEATURE_MAP.get(provider) or [],
+                        }
+                        for provider in instance.available_providers
+                    },
+                },
+                # lti_configuration
+                # name
+                # logo
```

Comment:
The MFE is going to need to worry about dimensions and styling anyway, so we may as well let it set the logo HREF too. I can also imagine us actually hosting the image file w/in the MFE, so I'm inclined to leave this up to the frontend.

This also gives us faster deploys, should the value change, since we aren't storing this in the DB.